core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/common/decorator/FactorySelectionSorter.java

```diff
@@ -0,0 +1,82 @@
+package ai.timefold.solver.core.impl.heuristic.selector.common.decorator;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import ai.timefold.solver.core.api.domain.common.ComparatorFactory;
+import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.score.director.ScoreDirector;
+import ai.timefold.solver.core.config.heuristic.selector.common.decorator.SelectionSorterOrder;
+import ai.timefold.solver.core.impl.heuristic.move.Move;
+import ai.timefold.solver.core.impl.heuristic.selector.Selector;
+
+/**
+ * Sorts a selection {@link List} based on a {@link ComparatorFactory}.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <T> the selection type
+ */
+public final class FactorySelectionSorter<Solution_, T> implements SelectionSorter<Solution_, T> {
+
+    private final ComparatorFactory<Solution_, T, ? extends Comparable<?>> selectionComparatorFactory;
+    private final Comparator<Comparable<?>> appliedComparator;
+
+    public FactorySelectionSorter(ComparatorFactory<Solution_, T, ? extends Comparable<?>> selectionComparatorFactory,
+            SelectionSorterOrder selectionSorterOrder) {
+        this.selectionComparatorFactory = selectionComparatorFactory;
+        switch (selectionSorterOrder) {
+            case ASCENDING:
+                this.appliedComparator = (Comparator<Comparable<?>>) Comparator.naturalOrder();
+                break;
+            case DESCENDING:
+                this.appliedComparator = Collections.reverseOrder();
+                break;
+            default:
+                throw new IllegalStateException(
+                        "The selectionSorterOrder (%s) is not implemented.".formatted(selectionSorterOrder));
+        }
+    }
+
+    @Override
+    public void sort(ScoreDirector<Solution_> scoreDirector, List<T> selectionList) {
```

Comment:
Seems unnecessary. It propagates the `ScoreDirector` type to an entirely unrelated package of classes, only for the calling code to avoid saying `getWorkingSolution()`. IMO should be removed.

Reply:
This class has been modified, and this method no longer exists.